cluster-autoscaler/simulator/framework/handle.go

```diff
@@ -50,7 +55,9 @@ func NewHandle(informerFactory informers.SharedInformerFactory, schedConfig *sch
 	}
 	sharedLister := NewDelegatingSchedulerSharedLister()
 
-	schedulermetrics.InitMetrics()
+	initMetricsOnce.Do(func() {
```

Comment:
just because I have to ask...

Is `RegisterMetrics` the only flow that relies upon these global metrics vars? Is it feasible to refactor that so that no global vars are involved?

Reply:
I'll LGTM this commit to unblock the fix, @towca please respond with a follow-up if needed.

Reply:
> Is RegisterMetrics the only flow that relies upon these global metrics vars? Is it feasible to refactor that so that no global vars are involved?

Yes, I think so. In any case, this is on the framework side so there's no way to get around it in CA:
* `schdulerframework.NewFramework()` at the very end calls `setInstrumentedPlugins()`, which calls methods on global metric variables: https://github.com/kubernetes/kubernetes/blob/a27498ef42f27e7cddadb2c5b31bc80fed81d964/pkg/scheduler/framework/runtime/framework.go#L410.
* `schedulermetrics.InitMetrics()` initializes these global metric variables. `schedulermetrics.Register()` calls `InitMetrics()`, and then registers the metrics to be exposed from the binary - all in a `sync.Once.Do()` block: https://github.com/kubernetes/kubernetes/blob/a27498ef42f27e7cddadb2c5b31bc80fed81d964/pkg/scheduler/metrics/metrics.go#L133.
* In scheduler code, `InitMetrics()` is never used directly, both prod and test code just use `Register()` which only calls `InitMetrics()` once.
* In CA, we don't want to register the metrics, but we need to have them initialized so that the framework doesn't panic. So we need to call `InitMetrics()` without `Register()`, and lose the `sync.Once.Do()` that `Register()` has.
* Ideally, `InitMetrics()` would probably also use `sync.Once.Do()` internally so we wouldn't have to do that ourselves in CA. But I think it should be safe, it matches what scheduler is doing in `Register()`.